src/server/dfly_main.cc

```diff
@@ -547,7 +547,19 @@ error_code GetCGroupPath(string* memory_path, string* cpu_path) {
 }
 
 void UpdateResourceLimitsIfInsideContainer(io::MemInfoData* mdata, size_t* max_threads) {
-  using io::Exists;
+  using absl::StrCat;
+
+  auto read_mem = [](string_view path, size_t* output) {
+    auto file = io::ReadFileToString(path);
+    DVLOG(1) << "container limits: read " << path << ": " << file.value_or("N/A");
+
+    size_t temp = numeric_limits<size_t>::max();
+
+    if (file.has_value() && !absl::StartsWith(*file, "max"))
+      CHECK(absl::SimpleAtoi(*file, &temp));
```

Comment:
@talbii  given our past experience with parsing cgroups files, I would like to avoid using check-fails in this code and just give up on auto-deducing memory with printing errors where needed.

Reply:
great idea!